TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Security/Authorization/Privilege/Node/CreateNodePrivilege.php

```diff
@@ -38,34 +38,53 @@ class CreateNodePrivilege extends AbstractNodePrivilege
      */
     public function matchesSubject(PrivilegeSubjectInterface $subject)
     {
-        if ($subject instanceof CreateNodePrivilegeSubject === false && $subject instanceof MethodPrivilegeSubject === false) {
-            throw new InvalidPrivilegeTypeException(sprintf('Privileges of type "%s" only support subjects of type "%s" or "%s", but we got a subject of type: "%s".', CreateNodePrivilege::class, CreateNodePrivilegeSubject::class, MethodPrivilegeSubject::class, get_class($subject)), 1417014353);
+        if (!($subject instanceof CreateNodePrivilegeSubject) && !($subject instanceof MethodPrivilegeSubject)) {
+            throw new InvalidPrivilegeTypeException(
+                sprintf(
+                    'Privileges of type "%s" only support subjects of type "%s" or "%s", but we got a subject of type: "%s".',
+                    CreateNodePrivilege::class,
+                    CreateNodePrivilegeSubject::class,
+                    MethodPrivilegeSubject::class,
+                    \get_class($subject)
+                ),
+                1417014353
+            );
         }
 
         $this->initialize();
+        $allowedCreationNodeTypes = $this->nodeContext->getCreationNodeTypes();
         if ($subject instanceof MethodPrivilegeSubject) {
-            if ($this->methodPrivilege->matchesSubject($subject) === false) {
+            if (!$this->methodPrivilege->matchesSubject($subject)) {
                 return false;
             }
 
             $joinPoint = $subject->getJoinPoint();
-            $allowedCreationNodeTypes = $this->nodeContext->getCreationNodeTypes();
-            $actualNodeType = $joinPoint->getMethodName() === 'createNodeFromTemplate' ? $joinPoint->getMethodArgument('nodeTemplate')->getNodeType()->getName() : $joinPoint->getMethodArgument('nodeType')->getName();
+            $actualNodeType = $joinPoint->getMethodName() === 'createNodeFromTemplate'
+                ? $joinPoint->getMethodArgument('nodeTemplate')->getNodeType()->getName()
+                : $joinPoint->getMethodArgument('nodeType')->getName()
+            ;
 
-            if ($allowedCreationNodeTypes !== array() && !in_array($actualNodeType, $allowedCreationNodeTypes)) {
+            if ($allowedCreationNodeTypes !== [] && !\in_array($actualNodeType, $allowedCreationNodeTypes, true)) {
```

Comment:
`\in_array` => `in_array`

Reply:
see https://github.com/Roave/FunctionFQNReplacer#rationale why I did that, but I see the code style issue ;)

Reply:
fair point, but it's not really related to the change and if we go for the FQN for core functions we should do so everywhere (and personally I'm not convinced as it makes the code a lot less readable if you apply it everywhere)